### PR TITLE
Feature/pelagos 4751 symfony command to find files in error

### PR DESCRIPTION
**Pull request by fightingtexasaggie**

### CODE REVIEW

File: src/Command/PelagosClearErrorAndFileCommand.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+        // This section is just added for extra safety. Don't delete
+        // unless the file is represented in the datastore. Don't delete
+        // the only copy of a file.
+        $newUdi = preg_replace('/:/', '.', $udi);
+        $uploadDir = $this->homedirPrefix . '/upload/files';
+        $dqlSafe = "SELECT f FROM \App\Entity\File f
+            WHERE f.id <> :fileId
+            AND f.fileset = :filesetId
+            AND f.physicalFilePath like '$newUdi%'
```
Comment (fightingtexasaggie) [inline src/Command/PelagosClearErrorAndFileCommand.php:104]: dangerous! Parameterize this correctly!!!

File: src/Command/PelagosClearErrorAndFileCommand.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            AND f.physicalFilePath not like '$uploadDir/%'";
```
Comment (fightingtexasaggie) [inline src/Command/PelagosClearErrorAndFileCommand.php:105]: this one too!

File: src/Util/Datastore.php [hunk 3]
```diff
@@ -114,6 +114,7 @@ public function deleteFile(string $filePath, bool $deleteDir = false): bool
         if ($deleteDir) {
              $deleteFile = $this->deleteDir($filePath);
         } else {
+            echo("deleting:" . $filePath . "\n");
```
Comment (fightingtexasaggie) [inline src/Util/Datastore.php:117]: remove debug code

File: src/Command/PelagosCheckForErrorStatusCommand.php [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    protected function smartSha256Hash($filename)
+    {
+        $hash = $this->memcached->get($filename);
+        if ($hash) {
+            return $hash;
+        } else {
+            // I hardcoded to sha256 otherwise hashtype would have to be part
+            // of what is stored in cache. Named method accordingly.
+            $hash = hash_file("sha256", $filename);
+            $this->memcached->set($filename, $hash);
```
Comment (praneethpr) [inline src/Command/PelagosCheckForErrorStatusCommand.php:195]: Shouldn't the hash be returned here

**Review by praneethpr — APPROVED** [review]
LGTM!

**Review by mickel1138 — APPROVED** [review]
This is a fine tool

**Pull request merged**